15314f1ddb882700dc9fabc5ca961943/update/sysauto_script_3f17190097a00a10b7daf896f053afd2.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update table="sysauto_script">
+    <sysauto_script action="INSERT_OR_UPDATE">
+        <active>true</active>
+        <business_calendar/>
+        <condition/>
+        <conditional>false</conditional>
+        <entered_time>1970-01-01 00:00:00</entered_time>
+        <max_drift/>
+        <name>Fetch Label Dimensions</name>
+        <offset/>
+        <offset_type>0</offset_type>
+        <run_as display_value="System Administrator">6816f79cc0a8016401c5a33be04be441</run_as>
+        <run_as_tz/>
+        <run_dayofmonth>1</run_dayofmonth>
+        <run_dayofweek>1</run_dayofweek>
+        <run_period/>
+        <run_start>2024-02-19 07:19:44</run_start>
+        <run_time>1970-01-01 00:00:00</run_time>
+        <run_type>daily</run_type>
+        <script><![CDATA[try {
+	gs.info("Started fetching label dimensions from PCE");
+	var labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
+	labelDimensionGr.addNotNullQuery("href");
+	labelDimensionGr.query();
+	labelDimensionGr.setValue("isdeleted", true);
+	labelDimensionGr.setWorkflow(false);
+	labelDimensionGr.updateMultiple();
+	
+	// Fetch all the label dimensions from Illumio PCE
+	var utils = new x_illu2_illumio.IllumioUtils();
+	utils.fetchLabelDimensionsFromPCE();
+	
+	// Retry failed Label Dimension creations on Illumio PCE
+	utils.retryFailedLabelDimensionCreation();
+	
+	// Delete the records that aren't needed
+	labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
```

Comment:
there isn't a necessity to create another object when one is created on line 22 right?

Reply:
We’d have to create a new record as here we’re trying to clear out the previous GlideRecord session and start a new once. Hence, instead of defining a new variable we’ve reused a variable and reinitialized it.